packages/workbox-background-sync/src/lib/QueueDb.ts

```diff
@@ -0,0 +1,149 @@
+/*
+  Copyright 2018 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+import {openDB, DBSchema, IDBPDatabase} from 'idb';
+import {RequestData} from './StorableRequest.js';
+
+interface QueueDBSchema extends DBSchema {
+  requests: {
+    key: number;
+    value: UnidentifiedQueueStoreEntry;
+    indexes: {queueName: string};
+  };
+}
+
+const DB_VERSION = 3;
+const DB_NAME = 'workbox-background-sync';
+const OBJECT_STORE_NAME = 'requests';
+const INDEXED_PROP = 'queueName';
+
+export interface UnidentifiedQueueStoreEntry {
+  requestData: RequestData;
+  timestamp: number;
+  id?: number;
+  queueName?: string;
+  metadata?: object;
```

Comment:
I know that this is what the old TypeScript definition used, but maybe there's an opportunity to clean that up a bit to remove some of the ambiguity, since `object` is overly broad: https://2ality.com/2020/01/typing-objects-typescript.html#object-vs.-object-in-typescript

Something like `Record<string, any>` might make more sense, or something more strict if we know that the values of each property will only be a certain subset of types.

(I'm assuming we can making "breaking" changes like that for this definition since it's internal-only, and not part of the public API, but correct me if I'm wrong.)

Reply:
We talked about this a bit, but we ultimately expose it as public as part of the API: https://developers.google.com/web/tools/workbox/reference-docs/latest/module-workbox-background-sync.Queue#pushRequest so won't change it for now.

Reply:
👍